### PR TITLE
[s] Fixes an href exploit related to shuttle docking

### LINKED ISSUE #47142: You can fly shuttles anywhere
Href exploit, you can forge the shuttle dock ID with any shuttle console to send them to any other unoccupied dock.

### Reproduction:

- Find a shuttle console
- Grab the destination selection href
- Swap the navigation ID with others that you probably shouldn't know
- BYOND TAKE THE WHEEL
- Get hated by both Centcom and the Syndicate, great job.

SpessTheFinalFrontier

> Spess, the final frontier. These are the voyages of the local assistant Fr--- "STOP GOOFING WITH THE SHUTTLE AND DO YOUR JOB"

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

As title
closes #47142 

## Changelog
:cl:
fix: Fixed an href exploit related to shuttle docking
/:cl:

### CODE REVIEW

File: code/modules/shuttle/computer.dm [hunk 1]
```diff
@@ -57,6 +57,11 @@
 			if(M.mode != SHUTTLE_IDLE)
 				to_chat(usr, "<span class='warning'>Shuttle already in transit.</span>")
 				return
+		if(!(href_list["move"] in params2list(possible_destinations)))
+			to_chat(usr, "<span class='userdanger'>I'm sure your \"Special\" sock would be proud of you for this action.</span>") // thanks citruses
```
Comment (AnturK) [inline code/modules/shuttle/computer.dm]: No.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/computer.dm (hunk 1 above)
Comment (actioninja) [reply code/modules/shuttle/computer.dm]: fun police

File: code/modules/shuttle/computer.dm [hunk 2]
```diff
@@ -58,7 +58,6 @@
 				to_chat(usr, "<span class='warning'>Shuttle already in transit.</span>")
 				return
 		if(!(href_list["move"] in params2list(possible_destinations)))
-			to_chat(usr, "<span class='userdanger'>I'm sure your \"Special\" sock would be proud of you for this action.</span>") // thanks citruses
```
Comment (GravitronBlue) [inline code/modules/shuttle/computer.dm]: 👀

File: code/modules/shuttle/computer.dm (hunk 1 above)
Comment (GravitronBlue) [reply code/modules/shuttle/computer.dm]: Wsock reference?
Personally don't need client-side hacks to find href exploits like this, nor do I support griefing when it isn't at least funny, which is mostly why I'm trying to find and get abusable hrefs fixed.

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**